### PR TITLE
[MB-4557] added addl error handling into support api

### PR DESCRIPTION
**Pull request by pearl-truss**
## Description

This PR updates the payment request support api to process the more specific types of errors coming out of the EDI generator

## Review Notes

I realize in `payment_request.go` there are 2 similar switch statements for handling the different types of errors. To be more DRY I tried to break it out into a separate function. However, I was unable to since the case statement returns different types of errors with custom types when defining the function there was no type I could specify for the expected return. If there is a way to encapsulate the switch statement that I've missed, please let me know and I'll update!

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-4557) for this change

### CODE REVIEW

**Review by shimonacarvalho — COMMENTED** [review]
Here's a potential way to fix the switch statement.

File: pkg/handlers/supportapi/payment_request.go [hunk 1]
```diff
@@ -192,13 +192,63 @@ func (h GetPaymentRequestEDIHandler) Handle(params paymentrequestop.GetPaymentRe
 	edi858c, err := h.GHCPaymentRequestInvoiceGenerator.Generate(paymentRequest, false)
 	if err != nil {
 		logger.Error(fmt.Sprintf("Error generating EDI segments for payment request ID: %s: %s", paymentRequestID, err))
-		return paymentrequestop.NewGetPaymentRequestEDIInternalServerError().WithPayload(payloads.InternalServerError(handlers.FmtString(err.Error()), h.GetTraceID()))
+		switch e := err.(type) {
```
[the remark this answers is not in the document]
Comment (pearl-truss) [reply pkg/handlers/supportapi/payment_request.go]: Thanks @shimonacarvalho, that's way better than having 2 repeating switch statement!

**Review by gidjin — APPROVED** [review]
Tested it out locally I'm getting errors as expected when calling the support endpoint for generating an edi. One nit, but otherwise 🚀

File: pkg/handlers/supportapi/payment_request.go [hunk 2]
```diff
@@ -190,15 +190,37 @@ func (h GetPaymentRequestEDIHandler) Handle(params paymentrequestop.GetPaymentRe
...
 	edi858c, err := h.GHCPaymentRequestInvoiceGenerator.Generate(paymentRequest, false)
-	if err != nil {
-		logger.Error(fmt.Sprintf("Error generating EDI segments for payment request ID: %s: %s", paymentRequestID, err))
-		return paymentrequestop.NewGetPaymentRequestEDIInternalServerError().WithPayload(payloads.InternalServerError(handlers.FmtString(err.Error()), h.GetTraceID()))
+	if err == nil {
+		payload.Edi, err = edi858c.EDIString()
 	}
-
-	payload.Edi, err = edi858c.EDIString()
 	if err != nil {
```
[the remark this answers is not in the document]
Comment (pearl-truss) [reply pkg/handlers/supportapi/payment_request.go:196]: @gidjin the two if statements follow an existing pattern @shimonacarvalho mentioned above. "// Don't use else here bc you want this to run if either First or Second thing failed."

https://github.com/transcom/mymove/pull/5074#discussion_r511045925

[the remark this answers is not in the document]
Comment (gidjin) [reply pkg/handlers/supportapi/payment_request.go:196]: Ah I missed the reuse of err inside the body of the first if.

**Pull request merged**